### ISSUE TITLE
Not able to open Xplorer on Windows 7

### ISSUE THREAD

**Issue by im-coder-lg** [kimlimjustin/xplorer#69]
**Describe the bug**
So, I installed Xplorer today and tried launching it. There is no error message, no window of Xplorer. All it opens is multiple instances of my VS Code and loading my computer up. I have attached screenshots below, can you try guessing the error? 

**To Reproduce**
Steps to reproduce the behaviour:
1. Install Xplorer
2. Launch it
3. It opens VS Code rather than Xplorer's window

**Expected behavior**
Xplorer to open

**Screenshots**
I will add screenshots+video afterwards, I don't have enough time as of now but will post it soon.

**Desktop (please complete the following information):**
 - OS Version: Windows 7 Home Premium and another computer with Windows 7 Ultimate(both have errors
 - App Version: v0.1.0

**Additional comment**
Add any other comment about the problem here. Does this problem happen again after restarting the app?

Any number of times, whenever I launch Xplorer, it does these:
1. Open VS Code
2. Add 3+ processes to my computer
3. Won't let me close them easily
4. Load my computer up and get it to almost the verge of bursting.

My computer has 4 GB of memory. Does Xplorer really need almost the entire CPU to work(genuine question, no offence and no sarcasm)?

**Comment by kimlimjustin** [comment]
Which architecture does your computer use? 32 bit or 64 bit? 

> My computer has 4 GB of memory. Does Xplorer really need almost the entire CPU to work(genuine question, no offence and no sarcasm)?
Xplorer should be able to work if VScode does

**Comment by im-coder-lg** [comment]
x64, and moreover, on the Home Premium PC, VS code loads it up but on the other one, it doesn't.

**Comment by kimlimjustin** [comment]
> x64, and moreover, on the Home Premium PC, VS code loads it up but on the other one, it doesn't.

Erm did you try to restart it and open it again? Also, for the problem of Xplorer opening VSCode, I think this is a bug of VSCode, because Xplorer needs to detect the existence of VSCode by running `code --version`, if this goes correctly, it is supposed to log the version of VSCode without opening VSCode. Erm, is your VSCode up to date? If yes, I'll try to repro this and figure out why is it

**Comment by im-coder-lg** [comment]
I mentioned in the description that it doesn't work even if I close and open it. It just loads the computer up. I think I will have to check my system variables, since I don't remember adding VS Code to it and using the `code` command in this PC.

**Comment by im-coder-lg** [comment]
Update: I did add it but the version doesn't show up, maybe Xplorer is trying to find the version. Let me try again later.

**Comment by kimlimjustin** [comment]
VSCode will auto add the `code` command to your PC, I'm not sure if the VSCode still supports Windows 7 or not since Microsoft announced the end of the windows 7 life and Microsoft developed VSCode...

I think I'll just refactor the way of finding the existence of VSCode in the future version and if the exe doesn't work. Could you please consider build from the source?

**Comment by kimlimjustin** [comment]
Oh, the documentation of Electron said that Windows 7 is still supported.

> Windows 7 and later are supported, older operating systems are not supported (and do not work).

Could you consider build from source (It requires Visual Studio C++ tho)? If you are not available, I can do it myself, but can't do really soon, still got so many things to do (and school xD)

**Comment by im-coder-lg** [comment]
@kimlimjustin they didn't end Windows 7 support. Last week, I installed VS 
Code 1.60.2 on my Windows 7 Ultimate PC, and it is build 7601.

**Comment by kimlimjustin** [comment]
Ye, I realized it, could you consider build from the source?

**Comment by im-coder-lg** [comment]
I did. I tried cloning and building but it didn't work. I even tried a Gitpod virtual environment and tried triggering a Windows build but it didn't work.

**Comment by kimlimjustin** [comment]
> I did. I tried cloning and building but it didn't work. I even tried a Gitpod virtual environment and tried triggering a Windows build but it didn't work.

Gitpod won't work because this is electron, any error when building ?

**Comment by im-coder-lg** [comment]
There was, and Gitpod will work since I did this method: yarn build:win
Moreover, I configured the workspace to a VNC one so that I can use Gitpod 
freely. You can see everything here:

A Gitpod Dockerfile for the image(it runs on VMs but uses images)
A Gitpod Config for configuration.
That's it.

**Comment by im-coder-lg** [comment]
And moreover, there is another project called Google Assistant Unofficial Desktop Client, it uses Electron and there is a PR on hold that adds extensive Gitpod Support. It is on hold because it will be merged after the next release. But it uses Electron and has Gitpod Support. Maybe there is some code error somewhere or something like that.

**Comment by im-coder-lg** [comment]
The old comment I left via mail was outdated. What I tried to say was:

There was, and Gitpod will work since I did this method: yarn build:win
Moreover, I configured the workspace to a VNC one so that I can use Gitpod freely. You can see everything here:

A Gitpod Dockerfile for the image(it runs on VMs but uses images)
https://github.com/im-coder-lg/xplorer/blob/d6038b10ba6388ff8b69ecb70ed17d21d337e3e6/.gitpod.Dockerfile#L1-L7
A Gitpod Config for configuration.
https://github.com/im-coder-lg/xplorer/blob/d6038b10ba6388ff8b69ecb70ed17d21d337e3e6/.gitpod.yml#L1-L15
That's it.

**Comment by kimlimjustin** [comment]
To run the development of Xplorer on Windows, you'll need Visual Studio C++, I think you need to config that first.

**Comment by im-coder-lg** [comment]
I don't have enough space for Visual Studio itself 😂 my PC has only 10 GB left, there is ABSOLUTELY no use of clearing because it somehow fills it up and I am not going to install Visual Studio C++ on my Windows 10 PC because same thing, not enough storage! But if there is a way to fix this from Gitpod, I'd like that since my main IDE is Gitpod, I rarely use the local VS Code.

**Comment by im-coder-lg** [comment]
Is normal C++ enough?

**Comment by im-coder-lg** [comment]
Or I have to get the one Microsoft offers?

**Comment by kimlimjustin** [comment]
Nope, it needs the `Desktop development with C++` to run (around 5gb)

**Comment by im-coder-lg** [comment]
Ah damn! I have to try building on another computer, which I absolutely 
hate. I need two computers to be used simultaneously which is irritating 
but I will have to check that. As a matter of curiosity, why do I need C++? 
I mean, since the app is made from TypeScript, why do I need C++ here?

**Comment by kimlimjustin** [comment]
> Ah damn! I have to try building on another computer, which I absolutely hate. I need two computers to be used simultaneously which is irritating but I will have to check that. As a matter of curiosity, why do I need C++? I mean, since the app is made from TypeScript, why do I need C++ here?

This is because there're some features that are only supported on Windows, such as acrylic window (tho IDK if is this gonna work on win 7) and extract exe icon, this is done via C++ bindings....

**Comment by im-coder-lg** [comment]
Okay... Thanks for explaining that! I think I will try installing C++ later since I am busy as of now.

**Comment by kimlimjustin** [comment]
okay

**Comment by kimlimjustin** [comment]
For those who want to try Xplorer in the future, please use the GitPod,  closing this for now.......
